ElevatorSim/src/elevatorsim/floor/FloorController.java

```diff
@@ -34,9 +35,16 @@ public FloorController(String name, HashMap<Integer, Floor> floors, List<Message
 	 */
 	public void run() {
 		Scheduler scheduler = Scheduler.getInstance();
-		for(MessageRequest request : requests) {
+		
+		for(MessageRequest request : requests.values()) {
```

Comment:
So this is an interesting way to do this. Looks like you are iterating all the messages and throwing them to the specified list in the floor. It's a bit different than I was thinking, where I was thinking we would send a list or map of all the requests to a specific floor at floor creation and let the floor constructor manage that. Just some thoughts....

Reply:
@MichaelPatsula ?

Reply:
I think the current way is fine for iteration 1. Looks like it works good. Just something to consider moving forward in the future, so I thought I would bring it up.

Reply:
Do you think the scheduler should be notified at the floor creation time too, or just passed the necessary MessageRequests at floor creation time? If you were thinking the latter, the reason why I iterated through all the messages and threw them to their corresponding floor is the following:

1. The requests are ultimately used by the floor to activate or deactivate certain lamps, so I passed the request to the floor and the scheduler basically at the same time in an effort to be as realistic as possible.
2. I thought it would a little messy to be iterating over message requests twice, one in main, and a second time in the floor Controller.